spec/requests/api/v1/moves_controller_index_spec.rb

```diff
@@ -85,6 +85,87 @@
           expect(response_json['meta']['pagination']).to include_json(meta_pagination)
         end
       end
+
+      describe 'importing moves from NOMIS passing all required filters (NOMIS agency_id and date)' do
+        let(:from_location) { moves.first.from_location }
+        let(:date) { Date.today }
+        let(:filters) do
+          {
+            from_location_id: from_location.id,
+            date_from: date.to_s
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before do
+          allow(NomisClient::Moves).to receive(:get).and_return([])
+
+          moves_importer = instance_double('Moves::Importer', call: [])
+          allow(Moves::Importer).to receive(:new).and_return(moves_importer)
+
+          get '/api/v1/moves', headers: headers, params: params
+        end
+
+        it 'invokes the client library to fetch moves from NOMIS', skip_before: true do
+          expect(NomisClient::Moves).to have_received(:get).with(from_location.nomis_agency_id, date)
+        end
+
+        it 'invokes the service to create moves into the database', skip_before: true do
+          expect(Moves::Importer).to have_received(:new).with([])
+        end
+      end
+
+      describe 'not importing moves from NOMIS when missing filters' do
+        let(:from_location) { moves.first.from_location }
+        let(:filters) do
+          {
+            from_location_id: from_location.id
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before do
+          allow(NomisClient::Moves).to receive(:get).and_return([])
+
+          moves_importer = instance_double('Moves::Importer', call: [])
+          allow(Moves::Importer).to receive(:new).and_return(moves_importer)
+
+          get '/api/v1/moves', headers: headers, params: params
+        end
+
+        it 'does not invoke the client library to fetch moves from NOMIS', skip_before: true do
+          expect(NomisClient::Moves).not_to have_received(:get)
+        end
+
+        it 'does not invoke the service to create moves into the database', skip_before: true do
+          expect(Moves::Importer).not_to have_received(:new)
+        end
+      end
+
+      describe 'not importing moves from NOMIS when an error occurs in NOMIS' do
+        let(:from_location) { moves.first.from_location }
+        let(:date) { Date.today }
+        let(:filters) do
+          {
+            from_location_id: from_location.id,
+            date_from: date.to_s
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before do
+          allow(NomisClient::Moves).to receive(:get).and_raise(StandardError)
```

Comment:
Is this `before` block needed if we specify `skip_before: true` in the only test?

Reply:
I think this `skip_before` is a bit confusing, @stevehook didn't like it from the beginning and he might have been right. So `skip_before` is referred to the first `before` in the file, the one on top of everything where it checks for the `metadata[:skip_before]` and if it's present it doesn't make the request. I use it to have a default `before` where we make the request, and in the few exceptions where I set doubles and method call expectations, I have to call it manually.

Reply:
Got it thanks. I've seen other usage of `skip_before` in the past which is a bit confusing.